LvivCompany.Bookstore/LvivCompany.Bookstore.DataAccess/Repo/BookRepository.cs

```diff
@@ -31,15 +33,15 @@ public override Task<Book> GetAsync(long id)
               .FirstOrDefaultAsync(x => x.Id == id);
         }
 
-        public async Task<IEnumerable<Book>> GetBySearch(string text)
+        public override async Task<IEnumerable<Book>> Get(Expression<Func<Book, bool>> filter)
         {
-
-                IEnumerable<Book> result = (from books in context.Books where books.Name.Contains(text) select books)
+            IQueryable<Book> query = context.Set<Book>();
+            query = query.Where(filter)
                 .Include(x => x.Publisher)
                 .Include(x => x.Category)
                 .Include(x => x.BookAuthors)
                 .ThenInclude(x => x.Author);
-                return await Task.FromResult<IEnumerable<Book>>(result);
+            return await query.ToListAsync();
```

Comment:
I've added related comment to @stepaHricishin PR #49 Please try refactoring your repos to reduce extra calls to `ToList`